app/Http/Controllers/AuthGoogleController.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use Illuminate\Http\Request;
+use Tymon\JWTAuth\Exceptions\JWTException;
+use Tymon\JWTAuth\Facades\JWTAuth;
+use App\User;
+use Auth;
+use Socialite;
+
+class AuthGoogleController extends Controller
+{
+
+    public function redirect()
+    {
+        return Socialite::driver('google')->redirect();
+    }
+
+
+    public function callback()
+    {
+        try {
+            $googleUser = Socialite::driver('google')->user();
+            $existUser = User::where('email',$googleUser->email)->first();
```

Comment:
@mykeels  you're making me cry. Where is the space between the function arguments? 😭

Reply:
`$existUser` 😨

Reply:
Lmaooo ... In my defence, I got that from [this blog post](https://appdividend.com/2018/04/11/laravel-google-login-tutorial), and became too pre-occupied with making it work. 😂